osquery/utils/aws_util.h

```diff
@@ -201,6 +201,10 @@ Status makeAWSClient(std::shared_ptr<Client>& client,
   } else {
     client_config.region = region;
   }
+
+  // Setup any proxy options on the config
+  setAWSProxy(client_config);
+
   client = std::make_shared<Client>(
```

Comment:
Is this going to fail if the proxy is misconfigured? If so we need to handle it here.

Reply:
@fmanco The client creation will not fail if the proxy is misconfigured, but instead the failure would happen when a requests is actually performed. the failure would be caught [here](https://github.com/aws/aws-sdk-cpp/blob/master/aws-cpp-sdk-core/source/client/AWSClient.cpp#L247) and logged/returned [here](https://github.com/aws/aws-sdk-cpp/blob/master/aws-cpp-sdk-core/source/client/AWSClient.cpp#L580-L586). do you have a suggestions as to how this could be handled here?

Reply:
If client creation won't fail then and there isn't a good way to check whether it is okay before doing the request, then there isn't much we can do.